x-pack/platform/plugins/shared/onechat/server/routes/tools.ts

```diff
@@ -6,35 +6,162 @@
  */
 
 import { schema } from '@kbn/config-schema';
+import { ToolType } from '@kbn/onechat-common';
 import type { RouteDependencies } from './types';
 import { getHandlerWrapper } from './wrap_handler';
 import { toolToDescriptor } from '../services/tools/utils/tool_conversion';
-import type { ListToolsResponse } from '../../common/http_api/tools';
+import type {
+  ListToolsResponse,
+  GetToolResponse,
+  DeleteToolResponse,
+  CreateToolPayload,
+  UpdateToolPayload,
+  CreateToolResponse,
+  UpdateToolResponse,
+} from '../../common/http_api/tools';
 import { apiPrivileges } from '../../common/features';
+import {
+  configurationSchema as esqlConfigSchema,
+  configurationUpdateSchema as esqlConfigUpdateSchema,
+} from '../services/tools/esql/schemas';
 
 export function registerToolsRoutes({ router, getInternalServices, logger }: RouteDependencies) {
   const wrapHandler = getHandlerWrapper({ logger });
 
-  router.post(
+  // list tools API
+  router.get(
     {
-      path: '/internal/onechat/tools',
+      path: '/api/chat/tools',
       security: {
         authz: { requiredPrivileges: [apiPrivileges.readOnechat] },
       },
       validate: false,
     },
     wrapHandler(async (ctx, request, response) => {
       const { tools: toolService } = getInternalServices();
-      const registry = toolService.registry.asScopedPublicRegistry({ request });
+      const registry = await toolService.getRegistry({ request });
       const tools = await registry.list({});
       return response.ok<ListToolsResponse>({
         body: {
-          tools: tools.map(toolToDescriptor),
+          results: tools.map(toolToDescriptor),
         },
       });
     })
   );
 
+  // get tool by ID
+  router.get(
+    {
+      path: '/api/chat/tools/{id}',
+      security: {
+        authz: { requiredPrivileges: [apiPrivileges.readOnechat] },
+      },
+      validate: {
+        params: schema.object({
+          id: schema.string(),
+        }),
+      },
+    },
+    wrapHandler(async (ctx, request, response) => {
+      const { id } = request.params;
+      const { tools: toolService } = getInternalServices();
+      const registry = await toolService.getRegistry({ request });
+      const tool = await registry.get(id);
+      return response.ok<GetToolResponse>({
+        body: toolToDescriptor(tool),
+      });
+    })
+  );
+
+  // create tool
+  router.post(
+    {
+      path: '/api/chat/tools',
+      security: {
+        authz: { requiredPrivileges: [apiPrivileges.manageOnechat] },
+      },
+      validate: {
+        body: schema.object({
+          id: schema.string(),
+          type: schema.maybe(
```

Comment:
@pquentin suggested that type should be explicit here and I agree with him it makes sense we should be explicit, enforcing to be aware there are different types.

Reply:
https://github.com/elastic/kibana/pull/227452/commits/727748e0a5fcd278f9969cbc72b637239a664b23

Reply:
Thanks! The feedback was not on the tool creation API but on the other APIs, where `type` was everywhere without a good reason. Indeed, if there's one place where we should leave the type, it's in the body of this API. (To have proper OpenAPI generation, it would have to be in the URL, but we all agree that's not ideal.)